docs/geps/gep-01.rst

```diff
@@ -0,0 +1,178 @@
+
+
+Column names (a.k.a. "variables" in Stata)
+------------------------------------------
+
+We impose a hard limit of 15 characters for all column names that are part of the API,
+i.e., those that are an input to or an output of GETTSIM's main simulation functions.
+This is for the benefit of Stata users, who face a strict limit of 32 characters for
+their column names. Furthermore, where developers using other languages may store
+different experiments in different variables, Stata users' only chance to distinguish
+them is to append characters to the column names.
+
+If a column is only present for internal use, it starts with an underscore.
+
+Even though not implemented at the time of this writing, we plan to allow users to pass
+in English column names and get English column names back. Similarly, there will be a
+pure ASCII option in German (this should always be fulfilled in the English version). If
+there is demand, we could also support standardised variables like in `EUROMOD
+<https://www.euromod.ac.uk/>`_ or the `CNEF <https://cnef.ehe.osu.edu/data/>`_ standard.
+
+It might be useful to append an underscore plus one of {``j``, ``m``, ``w``, ``t``} to
+indicate the time unit.
+
+
+
+Parameters of the taxes and transfers system
+--------------------------------------------
+
+The structure of these parameters will be laid out in :ref:`gep-3`; we just note some
+general naming considerations here.
+
+* There is a hierarchical structure to these parameters in that each of them is
+  associated with a group (e.g., ``arbeitsl_geld``, ``kinderzuschlag``). These groups or
+  abbreviations thereof do not re-appear in the name of the parameter.
+
+
+
+Python Identifiers (Functions, Variables)
+-----------------------------------------
+
+Python identifiers should generally be in English, unless they refer to a specific law
+or set of laws, which is where the same reasoning applies as above.
+
+The length of a variable name tends to be proportional to its scope. In a list
+comprehension or a short loop, ``i`` might be an acceptable name for the running
+variable. Variables that are used at many different places should have descriptive
+names.
+
+The name of variables should reflect the content or meaning of the variable and not the
+type (i.e., int, dict, list, ...). As for column names and parameters, in some cases it
+might be useful to append an underscore plus one of {``j``, ``m``, ``w``, ``t``} to
+indicate the time unit.
+
+Across variations that include the same identifier, this identifier should not be
+changed, even if it leads to long variable names (e.g., ``kinderfreibetrag``,
+``zu_verst_e_ohne_kinderfreibetrag``). This makes searching for that identifier easier
+and less error-prone.
+
+Function names should contain a verb. Moreover, the length of a function name is
+typically inversely proportional to its scope. The public functions like maximize and
```

Comment:
I feel flattered that you copied from estimagic but `maximize` and `minimize` are not part of gettsim, so maybe not a good example ;-)

Reply:
Maybe I liked it because parts of that sounded very familiar :-). 

Yip, that part still needs change.